File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) 
    {
        unordered_map<int, int> freq;

        for(int num: nums)
            freq[num]++;

        for(auto const &pair: freq)
        {
            if(pair.second > 1)
                return true;
        }

        return false;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] res = new int[nums.length];

        for(int i=0; i<nums.length; i++) {
            int prod = 1;
            for(int j=0; j<nums.length; j++) {
                if(j == i)
                    continue;
                prod = prod * nums[j];
            }
            res[i] = prod;
        }

        return res;
    }
}  
